pkg/filters/options.go

```diff
@@ -0,0 +1,147 @@
+// A resource is considered unused if it meets the following conditions:
+//   - Its age (measured from the last modified time) is within the range specified by older-than and newer-than flags.
+//     If older-than or newer-than is zero, no age limit is applied.
+//     If both flags are set, an error is returned.
+//   - Its size (measured in bytes) is within the range specified by MinSize and MaxSize flags.
+//     If MinSize or MaxSize is zero, no size limit is applied.
+//   - It does not have any labels that match the ExcludeLabels flag. The ExcludeLabels flag supports '=', '==', and '!=' operators,
+//     and multiple label pairs can be separated by commas. For example, -l key1=value1,key2!=value2.
+type Options struct {
+	// OlderThan is the minimum age of the resources to be considered unused
+	OlderThan string
+	// NewerThan is the maximum age of the resources to be considered unused
+	NewerThan string
+	// ExcludeLabels is a label selector to exclude resources with matching labels
+	// IncludeLabels conflicts with it, and when setting IncludeLabels, ExcludeLabels is ignored and set to empty
+	ExcludeLabels string
+	// IncludeLabels is a label selector to include resources with matching labels
+	IncludeLabels string
+	// ExcludeNamespaces is a namespace selector to exclude resources in matching namespaces
+	// IncludeNamespaces conflicts with it, and when setting IncludeNamespaces, ExcludeNamespaces is ignored and set to empty
+	ExcludeNamespaces []string
+	// IncludeNamespaces is a namespace selector to include resources in matching namespaces
+	IncludeNamespaces []string
+
+	namespace []string
+	once      sync.Once
+}
+
+// NewFilterOptions returns a new FilterOptions instance with default values
+func NewFilterOptions() *Options {
+	return &Options{
+		OlderThan:     "",
+		NewerThan:     "",
+		ExcludeLabels: "",
+	}
+}
+
+// Validate makes sure provided values for FilterOptions are valid
+func (o *Options) Validate() error {
+	if _, err := labels.Parse(o.ExcludeLabels); err != nil {
+		return err
+	}
+	if _, err := labels.Parse(o.IncludeLabels); err != nil {
+		return err
+	}
+
+	// Parse the older-than flag value into a time.Duration value
+	if o.OlderThan != "" {
+		olderThan, err := time.ParseDuration(o.OlderThan)
+		if err != nil {
+			return err
+		}
+		if olderThan < 0 {
+			return errors.New("OlderThan must be a non-negative duration")
+		}
+	}
+
+	// Parse the newer-than flag value into a time.Duration value
+	if o.NewerThan != "" {
+		newerThan, err := time.ParseDuration(o.NewerThan)
+		if err != nil {
+			return err
+		}
+		if newerThan < 0 {
+			return errors.New("NewerThan must be a non-negative duration")
+		}
+	}
+
+	return nil
+}
+
+// Modify modifies the options
+func (o *Options) Modify() {
+	o.modifyLabels()
+}
+
+// Namespaces returns the namespaces, only called once
+func (o *Options) Namespaces(clientset kubernetes.Interface) []string {
+	o.once.Do(func() {
```

Comment:
nit: instead of `once` we could jut check if `o.namespce` is `nil`.

Reply:
Checking ns saves an extra assignment. The advantage of once is that it is only executed once and does not require additional checks. However, checking ns will still make multiple requests to the server in some cases, such as when the server is busy during a period of time.（Adding additional judgments increases complexity）

Reply:
Wouldn't we have the same issue if the server is busy even with the once implementation?

Reply:
There are still some differences. When checking multiple resources at the same time (deploy, rs...)
once: No matter whether there is a problem or not, it will only be accessed once.
check: If the deploy check fails and the namespace string is empty, the next resource will continue to request services.
Supportive tools should have minimal impact on service.